xonsh/history.py

```diff
@@ -680,39 +648,42 @@ def _hist_gc(ns, hist):
             continue
 
 
-_HIST_MAIN_ACTIONS = LazyObject(lambda: {
+@lazyobject
+def HIST_SESSIONS():
+    return ('session', 'xonsh', 'bash', 'zsh')
+
+@lazyobject
+def _HIST_MAIN_ACTIONS():
+    return {
     'show': _hist_show,
     'xonsh': _hist_show,
     'zsh': _hist_show,
     'bash': _hist_show,
     'session': _hist_show,
-    'all': _hist_show,
     'id': lambda ns, hist: print(hist.sessionid),
     'file': lambda ns, hist: print(hist.filename),
     'info': _hist_info,
     'diff': _dh_main_action,
     'gc': _hist_gc,
-    }, globals(), '_HIST_MAIN_ACTIONS')
+    }
 
 
-def _hist_main(hist, args):
-    """This implements the history CLI."""
-    if not args or args[0] in ['-r'] or ensure_int_or_slice(args[0]):
+def parse_args(args):
```

Comment:
Due to amalgamation, please don't have such generically named funtions.  Retain the use of `_hist` in the name.  Also, all public functions need a docstring